pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -0,0 +1,151 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"fmt"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	route "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	envoy_jwt "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/jwt_authn/v2alpha"
+	http_conn "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	"github.com/golang/protobuf/ptypes/empty"
+	"istio.io/api/security/v1beta1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+	"istio.io/istio/pilot/pkg/security/authn"
+	authn_model "istio.io/istio/pilot/pkg/security/model"
+	"istio.io/pkg/log"
+)
+
+// Implemenation of authn.PolicyApplier with v1beta1 API.
+type v1beta1PolicyApplier struct {
+	jwtPolicies []*model.Config
+	// TODO: add mTLS configs.
+	// TODO: add v1alpha1 fallback configs.
+}
+
+func (a v1beta1PolicyApplier) JwtFilter(isXDSMarshalingToAnyEnabled bool) *http_conn.HttpFilter {
+	var filterConfigProto *envoy_jwt.JwtAuthentication
+	filterConfigProto = convertToEnvoyJwtConfig(a.aggregateJWTRules())
+
+	if filterConfigProto == nil {
+		return nil
+	}
+	out := &http_conn.HttpFilter{
+		Name: authn_model.EnvoyJwtFilterName,
+	}
+	if isXDSMarshalingToAnyEnabled {
+		out.ConfigType = &http_conn.HttpFilter_TypedConfig{TypedConfig: util.MessageToAny(filterConfigProto)}
+	} else {
+		out.ConfigType = &http_conn.HttpFilter_Config{Config: util.MessageToStruct(filterConfigProto)}
+	}
+	return out
+}
+
+func (a v1beta1PolicyApplier) AuthNFilter(proxyType model.NodeType, isXDSMarshalingToAnyEnabled bool) *http_conn.HttpFilter {
+	// TODO(diemtvu) implement this.
+	log.Errorf("AuthNFilter(%v, %v) is not yet implemented", proxyType, isXDSMarshalingToAnyEnabled)
+	return nil
+}
+
+func (a v1beta1PolicyApplier) InboundFilterChain(sdsUdsPath string, meta *model.NodeMetadata) []plugin.FilterChain {
+	// TODO(diemtvu) implement this.
+	log.Errorf("InboundFilterChain is not yet implemented")
+	return nil
+}
+
+// NewPolicyApplier returns new applier for v1beta1 authentication policies.
+func NewPolicyApplier(jwtPolicies []*model.Config) authn.PolicyApplier {
+	return &v1beta1PolicyApplier{
+		jwtPolicies: jwtPolicies,
+	}
+}
+
+func (a v1beta1PolicyApplier) aggregateJWTRules() []*v1beta1.JWT {
+	ret := []*v1beta1.JWT{}
+	// TODO: deduplicate JWT with the same issuer ??
+	// TODO: sort the rule by issuer for consistency ??
```

Comment:
These should be sorted by creation date for consistency with other APIs. This is a must have -- the result of flip flopping config has resulting in many issues included traffic loss, OOM, etc

Reply:
Jwt policies are mergeable, i.e all rules are appended into one big list. Also, I'm thinking of having the sort in the push-context instead (which is the common convention for all other configs).

Reply:
Yeah we don't need to sort here, we just need to make sure there is consistency in the final result sent to envoy